### PR TITLE
Create a mongodb-objects service and a publications endpoint

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: lib/Service/ObjectService.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace OCA\OpenCatalog\Service;
+
+use GuzzleHttp\Client;
+use Symfony\Component\Uid\Uuid;
+
+class ObjectService
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:8]: Missing a class docblock

File: lib/Service/ObjectService.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+{
+
+	public const BASE_OBJECT = [
+		'database'   => 'objects',
+		'collection' => 'json',
+	];
+
+	private function getClient(array $config): Client
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:16]: Missing function docblockS

File: lib/Service/ObjectService.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		unset($guzzleConf['mongodbCluster']);
+
+		return new Client($config);
+	}
+	public function saveObject(array $data, array $config): array
+	{
+		$client = $this->getClient(config: $config);
+
+		$object 			      = self::BASE_OBJECT;
+		$object['dataSource']     = $config['mongodbCluster'];
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:28]: allignment^

File: lib/Service/ObjectService.php [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	];
+
+	private function getClient(array $config): Client
+	{
+		$guzzleConf = $config;
+		unset($guzzleConf['mongodbCluster']);
+
+		return new Client($config);
+	}
+	public function saveObject(array $data, array $config): array
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:23]: Missing empty line^

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

**Review by remko48 — APPROVED** [review]

**Pull request merged**